server.js

```diff
@@ -1,33 +1,115 @@
-import express from 'express'
-import bodyParser from 'body-parser'
-import cors from 'cors'
-
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
-const port = process.env.PORT || 8080
+// The main go-to for starting APIs in JS, in Node. Really handy, mature tool for handling JS in back-end. 
+// Type npm run dev to start
+import express, { request, response } from "express"
+const nominees = require('./data/golden-globes.json')
+// This variable will allow us to create a server, which is a program that allows us to listen to incoming transmissions on a port. 65,000 ports on the computer.
 const app = express()
+const port = process.env.PORT || 8080;
 
-// Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+// Gives it a port and a function. Listen on port 8080, and the function is "what should I do when it starts?"
+app.listen(port, () => {
+  console.log(`Hello console, the server is up and running on port ${port}.`)
+})
 
-// Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
+// 🔵 Root endpoint.
+app.get('/', (request, response) => {
+  // For every endpoint you have, you have to send a response at some point. 
+  response.send("Hello World! See available endpoints at: https://github.com/AxelHammarback/project-express-api")
 })
 
-// Start the server
-app.listen(port, () => {
-  console.log(`Server running on http://localhost:${port}`)
+// 🔵 Endpoint that lists all the nominees (since 2010 – that's when the data starts).
+app.get('/nominees', (request, response) => {
+  response.send(nominees)
+})
+
+// 🔵 Endpoint that lists all nominees for a search value – either actor or movie/series title (the 'nominee' field).
+// 🟠 Example URL: /nominees/search?search=leo
+app.get('/nominees/search', (request, response) => {
+  const search = request.query.search
+  // "search" is the variable with the value that's being searched for.
+  let filteredNominees = nominees.filter(
+    item => item.nominee.toLowerCase().includes(search.toLowerCase())
+  )
+  if (filteredNominees.length === 0) {
+    response.status(404).json(`We searched around the Globe, but found no matches for '${search}'.`)
+  } else {
+    response.send(filteredNominees)
+  }
+})
```

Comment:
Nice with a search option! Endpoints should be named after what they return and not how they are filtered, so this could be a query param directly under you `/nominees` endpoint:
`/nominees?search=avatar`